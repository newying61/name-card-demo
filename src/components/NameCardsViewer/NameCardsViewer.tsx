import React, { useState } from 'react';
import { NameCard } from '../../types/NameCardType';
import { Wrapper } from './NameCardsViewer.style';
import NameCardsList from '../NameCardsList';
import SearchBox from '../SearchBox';

interface NameCardsViewerProps {
  nameCards: NameCard[];
  history: any;
  deleteNameCard: (id: string) => void;
}

const NameCardsViewer: React.FC<NameCardsViewerProps> = ({ nameCards, history, deleteNameCard }) => {

  // TODO: Find out whether we need to debounce the input
  const [searchTerm, setSearchTerm] = useState('');

  const onEdit = (card: NameCard) => {
    history.push(`/cards/${card.id}`);
  }

  const onDelete = (card: NameCard) => {
    // TODO: Is it better to display a confirm screen??
    deleteNameCard(card.id);
  }

  // TODO: If search term is used somewhere else, consider move it into store
  const list = nameCards.filter(
    (card) => card.firstName.toLowerCase().indexOf(searchTerm) > -1
      || card.lastName.toLowerCase().indexOf(searchTerm) > -1);

  return (
    <Wrapper>
      <SearchBox searchTerm={searchTerm} handleChange={setSearchTerm} />
      <NameCardsList nameCards={list} onEdit={onEdit} onDelete={onDelete} />
    </Wrapper>
  );
}

export default NameCardsViewer;
