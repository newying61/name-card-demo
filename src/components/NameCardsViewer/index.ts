import NameCardsViewer from './NameCardsViewer';

export default NameCardsViewer;
