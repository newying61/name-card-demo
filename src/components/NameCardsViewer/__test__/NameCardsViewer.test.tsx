import React from 'react';
import { render, cleanup } from '@testing-library/react';
import NameCardsViewer from '../';

afterEach(cleanup);

describe('AppNav', () => {
    it('should render without crash', () => {
      const { container } = render(
        <NameCardsViewer nameCards={[]} history={{}} deleteNameCard={jest.fn()} />
      );

      expect(container.firstChild).toMatchSnapshot();
  });
});

// TODO: Add more tests
