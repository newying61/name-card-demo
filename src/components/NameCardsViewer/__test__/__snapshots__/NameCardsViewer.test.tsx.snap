// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNav should render without crash 1`] = `
<div
  class="sc-bdVaJa cAxzPl"
>
  <div
    class="form-group"
  >
    <label
      class="form-label"
      for="search-box"
    >
      Search by name
    </label>
    <input
      class="sc-gzVnrw kSznVF form-control"
      id="search-box"
      name="searchBox"
      placeholder=""
      type="text"
      value=""
    />
  </div>
  <div>
    No cards
  </div>
</div>
`;
